src/notifications/container.js

```diff
@@ -0,0 +1,246 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import { bindActionCreators } from 'redux'
+import Waypoint from 'react-waypoint'
+
+import { LoadingIndicator, ToggleSwitch } from 'src/common-ui/components'
+import * as actions from './actions'
+import * as selectors from './selectors'
+import NotificationList from './components/NotificationList'
+import Notification from './components/Notification'
+import StatusHeading from './components/StatusHeading'
+import ReadHeader from './components/ReadHeader'
+import OpenLinkButton from './components/OpenLinkButton'
+import ActionButton from './components/ActionButton'
+import { actionRegistry } from './registry'
+import OptIn from './components/OptIn'
+
+class NotificationContainer extends Component {
+    static propTypes = {
+        unreadNotificationList: PropTypes.arrayOf(PropTypes.object).isRequired,
+        readNotificationList: PropTypes.object.isRequired,
+        init: PropTypes.func.isRequired,
+        showMoreIndex: PropTypes.string,
+        handleToggleShowMore: PropTypes.func.isRequired,
+        handleTick: PropTypes.func.isRequired,
+        onBottomReached: PropTypes.func.isRequired,
+        isLoading: PropTypes.bool.isRequired,
+        needsWaypoint: PropTypes.bool.isRequired,
+        isReadExpanded: PropTypes.bool.isRequired,
+        toggleReadExpand: PropTypes.func.isRequired,
+        isReadShow: PropTypes.bool.isRequired,
+        messageCharLimit: PropTypes.number.isRequired,
+        shouldTrack: PropTypes.bool.isRequired,
+        setShouldTrack: PropTypes.func.isRequired,
+    }
+
+    static defaultProps = {
+        messageCharLimit: 150,
+    }
+
+    componentDidMount() {
+        this.props.init()
+    }
+
+    isNotificationTruncated(message) {
+        return message.length > this.props.messageCharLimit
+    }
+
+    truncateText(message) {
+        const lastSpaceBeforeCutoff = message.lastIndexOf(
+            ' ',
+            this.props.messageCharLimit,
+        )
+        const trunctatedText = message.substr(0, lastSpaceBeforeCutoff) + '...'
+        return trunctatedText
+    }
+
+    handleToggleStorageOption(action) {
+        const { shouldTrack } = this.props
+
+        action = {
+            ...action,
+            value: !shouldTrack,
+        }
+
+        this.props.setShouldTrack(!shouldTrack)
+        actionRegistry[action.type]({
+            definition: action,
+        })
+    }
+
+    renderButtons(buttons) {
+        const { shouldTrack } = this.props
+
+        if (!buttons) {
+            return null
+        }
+
+        return buttons.map((button, i) => {
+            const { action } = button
+
+            if (action.type === 'go-to-url') {
```

Comment:
Make sure to replace these string literals with your constants from `notifications/action-types` module